setup.py

```diff
@@ -24,8 +24,9 @@ def __init__(self, user=False):
             import pybind11
             pybind11
         except ImportError:
-            if call([sys.executable, '-m', 'pip', 'install', 'pybind11']):
-                raise RuntimeError('pybind11 install failed.')
+            cmd = run([sys.executable, '-m', 'pip', 'install', 'pybind11'], capture_output=True)
+            if cmd:
```

Comment:
I might be missing something here. But `cmd` is an instance of `subprocess.CompletedProcess`. Casting this to `bool` as is done here doesn't tell us anything about whether the `pybind11` call succeeded or failed. The earlier implementation looks correct, since `call` returns the exit code of the call (0 or non-zero), which can be cast to `bool`.

Reply:
Ah, good catch! How about:

```suggestion
            if cmd.returncode != 0:
```

That should match:
```python
[ins] In [15]: bool(call(["/usr/bin/false"]))
Out[15]: True

[ins] In [16]: bool(run(["/usr/bin/false"], capture_output=True).returncode != 0)
Out[16]: True
```